lib/widgets/nutrition/nutritional_diary_detail.dart

```diff
@@ -87,125 +87,53 @@ class NutritionDiaryTable extends StatelessWidget {
-          ],
-        ),
-        TableRow(
-          children: [
-            Padding(
-              padding: const EdgeInsets.symmetric(vertical: tablePadding),
-              child: Text(AppLocalizations.of(context).fat),
-            ),
-            Text(AppLocalizations.of(context).gValue(planned.fat.toStringAsFixed(0))),
-            Text(AppLocalizations.of(context).gValue(logged.fat.toStringAsFixed(0))),
-            Text((logged.fat - planned.fat).toStringAsFixed(0)),
-          ],
-        ),
-        TableRow(
-          children: [
-            Padding(
-              padding: const EdgeInsets.symmetric(vertical: tablePadding, horizontal: 12),
-              child: Text(AppLocalizations.of(context).saturatedFat),
-            ),
-            Text(AppLocalizations.of(context).gValue(planned.fatSaturated.toStringAsFixed(0))),
-            Text(AppLocalizations.of(context).gValue(logged.fatSaturated.toStringAsFixed(0))),
-            Text((logged.fatSaturated - planned.fatSaturated).toStringAsFixed(0)),
-          ],
-        ),
-        TableRow(
-          children: [
-            Padding(
-              padding: const EdgeInsets.symmetric(vertical: tablePadding),
-              child: Text(AppLocalizations.of(context).fiber),
-            ),
-            Text(AppLocalizations.of(context).gValue(planned.fiber.toStringAsFixed(0))),
-            Text(AppLocalizations.of(context).gValue(logged.fiber.toStringAsFixed(0))),
-            Text((logged.fiber - planned.fiber).toStringAsFixed(0)),
-          ],
-        ),
-        TableRow(
-          children: [
-            Padding(
-              padding: const EdgeInsets.symmetric(vertical: tablePadding),
-              child: Text(AppLocalizations.of(context).sodium),
-            ),
-            Text(AppLocalizations.of(context).gValue(planned.sodium.toStringAsFixed(0))),
-            Text(AppLocalizations.of(context).gValue(logged.sodium.toStringAsFixed(0))),
-            Text((logged.sodium - planned.sodium).toStringAsFixed(0)),
-          ],
-        ),
+        TableRow(children: [
+          columnHeader(true, loc.macronutrients),
+          columnHeader(false, loc.planned),
+          columnHeader(false, loc.logged),
+          columnHeader(false, loc.difference),
+        ]),
+        macroRow(0, false, loc.energy, (NutritionalValues nv) => nv.energy),
+        macroRow(0, true, loc.protein, (NutritionalValues nv) => nv.protein),
+        macroRow(0, true, loc.carbohydrates, (NutritionalValues nv) => nv.carbohydrates),
+        macroRow(1, true, loc.sugars, (NutritionalValues nv) => nv.carbohydratesSugar),
+        macroRow(0, true, loc.fat, (NutritionalValues nv) => nv.fat),
+        macroRow(1, true, loc.saturatedFat, (NutritionalValues nv) => nv.fatSaturated),
+        macroRow(0, true, loc.fiber, (NutritionalValues nv) => nv.fiber),
+        macroRow(0, true, loc.sodium, (NutritionalValues nv) => nv.sodium),
```

Comment:
FYI, while that doesn't apply in this widget, the flutter team recommends creating new widgets instead of using helper methods https://www.youtube.com/watch?v=IOyq-eTRhvo

Reply:
i do think that would make the code more verbose. with the helper method it's really compact and easy to read.

but yeah, we can try out that approach later. let's just merge this now before conflicts accumulate.

Reply:
sure. Plus this is only important if we have nested and complex widget trees, which is not the case here